### ISSUE TITLE
Add Maximum Power Point Tracking (MPPT)

### ISSUE THREAD

**Issue by AsaphHall** [nomad-hzb/autoperosol#11]
The goal of this issue is to include MPPT into the TFSCO.

**Comment by AsaphHall** [comment]
This is a first version of the MPPT class.
![grafik](https://github.com/user-attachments/assets/c38d2492-7ab4-4d47-94ba-ae3a11d8b4b4)

**Comment by RoteKekse** [comment]
nice

**Comment by AsaphHall** [comment]
Maximum power point tracking (is_a scan, provides_input_for some MPPT Analysis)

Participants:
- Photovoltaic device
- Light emission device

Qualities:
- electric potential
- electric potential setting datum
- voltage step size
- voltage step size setting datum
- duration
- duration setting datum
- light intensity setting datum
- current
- current measurement datum
- time measurement datum
- power

MPPT Analysis (is_a data transformation)

Qualities:
- time until 90% efficiency
- time until 80% efficiency

Output:
- time until 90% efficiency measurement datum
- time until 80% efficiency measurement datum


![grafik](https://github.com/user-attachments/assets/1e42553a-7026-4042-939d-3992ad0b2dc7)

**Comment by RoteKekse** [comment]
in jv 
![grafik](https://github.com/user-attachments/assets/b658649a-52c8-47f5-a93a-a11167c2e892)
 
we also used the class tabular data `iv cuve` this is missing here right?

**Comment by RoteKekse** [comment]
![grafik](https://github.com/user-attachments/assets/22b7a9cb-aece-433b-874a-bc6c428fc267)

but seeing it now i am not sure if `has_specified_input` is correct is there something like `contains`for that kind of an object relation?

**Comment by AsaphHall** [comment]
> ![grafik](https://private-user-images.githubusercontent.com/36420750/379664100-22b7a9cb-aece-433b-874a-bc6c428fc267.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3Mjk3NTYxNTYsIm5iZiI6MTcyOTc1NTg1NiwicGF0aCI6Ii8zNjQyMDc1MC8zNzk2NjQxMDAtMjJiN2E5Y2ItYWVjZS00MzNiLTg3NGEtYmM2YzQyOGZjMjY3LnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDEwMjQlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQxMDI0VDA3NDQxNlomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTk1YmZkOTg5N2Y3NzA5N2M0MDRmZDhlYzMwOTdiNzI5MzQzZWUwMzk1ZjZjZGZiOWI3Njc4MmM3OTg3ODkwMjkmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0In0.eq7iigmyDlcc_jHwx1u0jnsM7HXNAaFHky2UN9DF_pE)
> 
> but seeing it now i am not sure if `has_specified_input` is correct is there something like `contains`for that kind of an object relation?
For the mppt analysis i used 'maximum power point tracking' `provies_input_for` some 'mppt analysis'. As of now `provides_input_for` has no inverse relation in the tfsco.

**Comment by RoteKekse** [comment]
yes that makes sense, but the input itself is missing, so in jv measurement we have the class `iv curve` is a `tabular data

**Comment by AsaphHall** [comment]
Additional classes:
MPPT time series
![grafik](https://github.com/user-attachments/assets/19d2dede-f3e7-430f-bed1-18c40db627ef)

**Comment by RoteKekse** [comment]
i would put the `electric potential setting datum` as part of the time series

**Comment by AsaphHall** [comment]
The relation 'has specified input' between 'mppt time series' and 'power', 'time measurement datum' and 'current measurment datum' has been removed. The relation 'has part' between the entities remains.
![grafik](https://github.com/user-attachments/assets/450d2312-9adc-4385-ba92-be7ab5c44141)

**Comment by AsaphHall** [comment]
![grafik](https://github.com/user-attachments/assets/43654f2d-1e4a-4e02-84dc-663171df742d)

**Comment by AsaphHall** [comment]
![Image](https://github.com/user-attachments/assets/ed46cf3c-ea46-46e7-999a-fb6914cdd5ed)
